src/Core/Grid/Definition/Definition.php

```diff
@@ -141,4 +142,48 @@ public function getFilters()
     {
         return $this->filters;
     }
+
+    /**
+     * @param string $name
+     */
+    public function setName($name)
+    {
+        if (!is_string($name)) {
+            throw new InvalidDataException('Definition name should be a string.');
+        }
+
+        $this->name = $name;
+    }
+
+    /**
+     * @param ColumnCollectionInterface $columns
+     */
+    public function setColumns($columns)
```

Comment:
I don't need them to be part of the contract as we retrieve a Definition instance and not a DefinitionInterface.

Reply:
but shouldnt `Definition` be final as it implements `DefinitionInterface`?

Regarding `setName()`, i guess you should be able to change it, but with other setter, i dont know.. :/
Definition already has ColumnCollection set (in constructor when created), there is no need to be able to set it again, because if you want to change something in it, you can do using `$definition->getColumns()`.

Reply:
ok so the module and the docs should describe that: I keep the setter for the name.

Also, there is no link between implementing an interface and being final. The way I see it:
* implementing the interface means you have access to methods available into the interface, you'll see that in the end the DefinitionFactory create a concrete implementation of Definition class :) This is why it's ok to have a setter into our concrete implementation that is not part of contract because we never assume we have this setter on something that expect the interface.

* it is true that I'd use to make my final classes implement an interface to let developers use their own implementations: we're doing an extensible CMS and I'm removing overrides system
* for me, making my class final means "this class have and only have the behavior we expect for it, if you want to alter it, rely on the contract and not on our own"

this is really defensive programming, I try to stabilize a 10 years old software and I need to ensure that what we provide can't be broken :)